src/Microsoft.AspNetCore.Routing/Template/RoutePrecedence.cs

```diff
@@ -233,6 +236,12 @@ private static int ComputeInboundPrecedenceDigit(RoutePatternPathSegment pathSeg
             }
             else if (part is RoutePatternParameterPart parameterPart)
             {
+                // Parameter with a required value is matched as a literal
+                if (RequiredValueHelpers.TryGetRequiredValue(routePattern, parameterPart, out _))
```

Comment:
I've been focused on getting matching working. Have not tested yet whether link generation should do the same

Reply:
I think for outbound we should treat it like a parameter. Inbound seems right.